src/StreamResourceFactoryInterface.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Interop\Http\Factory;
+
+use Psr\Http\Message\StreamInterface;
+
+interface StreamResourceFactoryInterface
+{
+    /**
+     * Create a new stream from a resource.
+     *
+     * The resource MUST be readable and SHOULD be seekable. It MAY be writable.
+     *
+     * @param resource $resource
+     *
+     * @return StreamInterface
+     */
+    public function createStreamFromResource($resource);
```

Comment:
There is no [type hint for resources](http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration).